src/Target.h

```diff
@@ -139,7 +141,7 @@ struct Target {
      * OpenGL, because it is not capable of gpgpu, and is not
      * scheduled via Func::gpu_tile. */
     bool has_gpu_feature() const {
-        return has_feature(CUDA) || has_feature(OpenCL);
+        return has_feature(CUDA) || has_feature(OpenCL) || has_feature(Renderscript);
```

Comment:
has_gpu_feature means gpu_tile, gpu_blocks, and gpu_threads all work, and
that you can compute_at an input at Var::gpu_blocks() and have shared
memory happen. i.e. the cuda model. I don't believe that's the case with
opengl or renderscript.

On Wed, Apr 29, 2015 at 4:57 PM, Alexander Aprelev <notifications@github.com
> wrote:

> In src/Target.h
> <https://github.com/halide/Halide/pull/761#discussion_r29393729>:
>
> > @@ -139,7 +141,7 @@ struct Target {
> >       * OpenGL, because it is not capable of gpgpu, and is not
> >       * scheduled via Func::gpu_tile. */
> >      bool has_gpu_feature() const {
> > -        return has_feature(CUDA) || has_feature(OpenCL);
> > +        return has_feature(CUDA) || has_feature(OpenCL) || has_feature(Renderscript);
>
> I'm not sure why we won't allow gpu_ scheduling for Renderscript kernels?
> Currently Func::shader() uses gpu_blocks when it sets up scheduling for
> Renderscript.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/761/files#r29393729>.
>